go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy_controller.go

```diff
@@ -0,0 +1,509 @@
+
+	}
+
+	klog.Info("Setting up event handlers for Baseline Admin Network Policy")
+	// setup banp informers, listers, queue
+	c.banpLister = banpInformer.Lister()
+	c.banpCacheSynced = banpInformer.Informer().HasSynced
+	c.banpQueue = workqueue.NewNamedRateLimitingQueue(
+		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
+		"baselineAdminNetworkPolicy",
+	)
+	_, err = banpInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.onBANPAdd,
+		UpdateFunc: c.onBANPUpdate,
+		DeleteFunc: c.onBANPDelete,
+	}))
+	if err != nil {
+		return nil, fmt.Errorf("could not add Event Handler for banpInformer during admin network policy controller initialization, %w", err)
+	}
+
+	klog.Info("Setting up event handlers for Namespaces in Admin Network Policy controller")
+	c.anpNamespaceLister = namespaceInformer.Lister()
+	c.anpNamespaceSynced = namespaceInformer.Informer().HasSynced
+	c.anpNamespaceQueue = workqueue.NewNamedRateLimitingQueue(
+		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
+		"anpNamespaces",
+	)
+	_, err = namespaceInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.onANPNamespaceAdd,
+		UpdateFunc: c.onANPNamespaceUpdate,
+		DeleteFunc: c.onANPNamespaceDelete,
+	}))
+	if err != nil {
+		return nil, fmt.Errorf("could not add Event Handler for namespace Informer during admin network policy controller initialization, %w", err)
+	}
+
+	klog.Info("Setting up event handlers for Pods in Admin Network Policy controller")
+	c.anpPodLister = podInformer.Lister()
+	c.anpPodSynced = podInformer.Informer().HasSynced
+	c.anpPodQueue = workqueue.NewNamedRateLimitingQueue(
+		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
+		"anpPods",
+	)
+	_, err = podInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.onANPPodAdd,
+		UpdateFunc: c.onANPPodUpdate,
+		DeleteFunc: c.onANPPodDelete,
+	}))
+	if err != nil {
+		return nil, fmt.Errorf("could not add Event Handler for pod Informer during admin network policy controller initialization, %w", err)
+	}
+
+	// TODO(tssurya): We don't use recorder now but will add events in future iterations
+	c.eventRecorder = recorder
+
+	return c, nil
+}
+
+// Run will not return until stopCh is closed. workers determines how many
+// objects (pods, namespaces, anps, banps) will be handled in parallel.
+func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+
+	klog.Infof("Starting controller %s", c.controllerName)
+
+	// Wait for the caches to be synced
+	klog.Info("Waiting for informer caches to sync")
+	if !util.WaitForNamedCacheSyncWithTimeout(c.controllerName, stopCh, c.anpCacheSynced, c.banpCacheSynced, c.anpNamespaceSynced, c.anpPodSynced) {
```

Comment:
I saw this is also done in the factory `Start`, so I guess we don't need to wait for caches sync here?

Reply:
hmm again good point!
I am guessing you are talking about:
```
	if config.OVNKubernetesFeature.EnableAdminNetworkPolicy && wf.anpFactory != nil {
		wf.anpFactory.Start(wf.stopChan)
		for oType, synced := range waitForCacheSyncWithTimeout(wf.anpFactory, wf.stopChan) {
			if !synced {
				return fmt.Errorf("error in syncing cache for %v informer", oType)
			}
		}
	}
```
maybe? iFactory and anpFactory are both synced there like you mention, I literally just copy/pasted these bits ROFL, if its safe at the watch factory start stage for the sync to happen and we don't need to wait here, I could remove it.. no harm in also double checking here I guess.